### PR TITLE
Upgrading to support aws > 3.1 provider

### PR DESCRIPTION
**Pull request by jamengual**
## what
* upgrade aws provider to > 3.1

## why
* Upgrading to version 3 of aws provider across the board

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,10 +1,21 @@
 terraform {
-  required_version = ">= 0.12.0"
-
+  required_version = ">= 0.12"
   required_providers {
-    aws      = ">= 2.0"
-    template = "~> 2.0"
-    local    = "~> 1.2"
-    null     = "~> 2.0"
+    aws = {
```
Comment (RothAndrew) [inline versions.tf:4]: Doesn't this new format require Terraform version to be >= 0.13?

**Comment by tomasbackman** [comment]
Nomenclature
`=> 2.0` means anything 2.0 or bigger, including 3.x.
`~> 2.0` means anything bigger on that major release, so anything 2.x but not 3.x

So it looks like the module already supports 3.x (and 2.x) since the version constraints uses `=>` and that the changes will make it only work for 3.1 and up. Which maybe is good but it is more of "dropping support for 2" than "adding support for 3"

File: versions.tf (hunk 1 above)
Comment (tomasbackman) [reply versions.tf:4]: Yes the below is a terraform 0.13 thing, not aws provider 3.x

File: versions.tf [hunk 2]
```diff
@@ -1,10 +1,21 @@
 terraform {
-  required_version = ">= 0.12.0"
-
+  required_version = ">= 0.12"
```
Comment (RothAndrew) [inline versions.tf:2]: @jamengual this needs to be updated to `>= 0.12.26` due to the new required_providers format.

**Review by RothAndrew — CHANGES_REQUESTED** [review]

**Comment by RothAndrew** [comment]
This module already supports any version of the `aws` provider greater than or equal to v2.0 due to use of `=>` rather than `~>`. The other changes are useful but to keep things clean I will pull them out into another PR.

Closing this one due to already supporting what this PR wants done.

**Comment by RothAndrew** [comment]
:poke: @jamengual @tomasbackman

**Pull request closed without merging**